File: front-end/src/models/Stats.js
export class Stats {
    constructor(maxHealth, health, strength, defense, dexterity, speed, accuracy, stun, burn, poison) {
        this.maxHealth = maxHealth;
        this.health = health;
        this.strength = strength;
        this.defense = defense;
        this.dexterity = dexterity;
        this.speed = speed;
        this.accuracy = accuracy;
        this.stun = stun;
        this.burn = burn;
        this.poison = poison;
    }

    static fromDict(data) {
        return new Stats(
            data.maxHealth,
            data.health,
            data.strength,
            data.defense,
            data.dexterity,
            data.speed,
            data.accuracy,
            data.stun,
            data.burn,
            data.poison
        );
    }
}

File: frontend(BAD)/code/models/stats.py
from dataclasses import dataclass


@dataclass
class Stats:
    maxHealth: float
    health: float
    strength: float
    defense: float
    dexterity: float
    speed: float
    accuracy: float

    @classmethod
    def from_dict(cls, d):
        return cls(
            maxHealth=d['maxHealth'],
            health=d['health'],
            strength=d['strength'],
            defense=d['defense'],
            dexterity=d['dexterity'],
            speed=d['speed'],
            accuracy=d['accuracy'],
        )
